### ISSUE TITLE
C++ issues

### ISSUE THREAD

**Issue by whit2333** [JeffersonLab/clara-demo#3]
Since the opencv demo is not building, can you provide a  trivial example for c++? I have tried myself but I am not sure I have things totally in order.

I have the "could not start container" error but nothing else to really understand what the problem is.

```
Distribution  :    clara-cre-4.3.8
CLAS12 plugin :    coatjava-5.9.0

==========================================
               CLARA FE/DPE
==========================================
 Name             = 172.17.0.1%7140_java
 Session          = whit_clara
 Start time       = 2019-06-28 13:09:45
 Version          = 4.3
 Lang             = Java
 Pool size        = 2

 Proxy Host       = 172.17.0.1
 Proxy Port       = 7140
==========================================
=========================================
                 CLARA DPE
=========================================
 Name             = 172.17.0.1%7150_cpp
 Date             = 2019-06-28 13:09:46
 Version          = 4.3
 Lang             = C++

 Proxy Host       = 172.17.0.1
 Proxy Port       = 7150

 FrontEnd Host    = 172.17.0.1
 FrontEnd Port    = 7140
 FrontEnd Lang    = Java
=========================================
==========================================
            CLARA Orchestrator
==========================================
 Front-end          = 172.17.0.1%7140_java
 Start time         = 2019-06-28 13:09:48
 Threads            = 4

 Input directory    = /mnt/old_home/whit/projects/clara/clara_home/data/input
 Output directory   = /mnt/old_home/whit/projects/clara/clara_home/data/output
 Output file prefix = out_
 Number of files  = 1
==========================================
2019-06-28 13:09:49.179: Waiting for local node...
2019-06-28 13:09:49.332: Start processing on 172.17.0.1...
2019-06-28 13:09:49.333: Searching services in 172.17.0.1...
2019-06-28 13:09:49.349: Deploying services in 172.17.0.1...
2019-06-28 13:09:49: started container = 172.17.0.1%7140_java:whit
2019-06-28 13:09:50: basic_string::_M_create
2019-06-28 13:09:50: started service = 172.17.0.1%7140_java:whit:DataManager  pool_size = 1
2019-06-28 13:09:50: started service = 172.17.0.1%7140_java:whit:HipoToHipoReader  pool_size = 1
2019-06-28 13:09:50: started service = 172.17.0.1%7140_java:whit:HipoToHipoWriter  pool_size = 1
2019-06-28 13:09:51.972: Could not use 172.17.0.1 for processing:
could not start container = 172.17.0.1%7150_cpp:whit
^C2019-06-28 13:09:59: removed service = 172.17.0.1%7140_java:whit:HipoToHipoWriter
2019-06-28 13:09:59: removed service = 172.17.0.1%7140_java:whit:HipoToHipoReader
2019-06-28 13:09:59: removed service = 172.17.0.1%7140_java:whit:DataManager
2019-06-28 13:09:59: removed container = 172.17.0.1%7140_java:whit
2019-06-28 13:09:59: shutdown DPE
```

**Comment by smancill** [comment]
Clara C++ never left beta, and it is currently unmantained because I am the main C++ developer but I was left out of any collaboration with JLab.

Nevertheless I have been asked to take a look and take care of any possible bug, so I will investigate this, in an "open-source-ish" sort of way.

You are right, that's not a good error message, and failing to start a container shouldn't happen.
I'll try to figure out what's going on.

**Comment by whit2333** [comment]
I have a little (non functioning) demo setup here: 
https://github.com/whit2333/clara_cpp_test

**Comment by whit2333** [comment]
@smancill , I found a clue in the system logs:

```
[114293.735827] c_dpe[23799]: segfault at 8 ip 00007fb529508f93 sp 00007fb517ffe8c0 error 4 in libxmsg.so[7fb5294d1000+65000] ```

**Comment by smancill** [comment]
It should be fixed by JeffersonLab/xmsg-cpp@93dfd68.

Rebulding Clara C++ from source should bring latest xMsg and it should work.
Otherwise try downgrading GCC.
